eta/core/datasets.py

```diff
@@ -1842,9 +1847,11 @@ def build(self, path, description=None, pretty_print=False,
                 data_method=data_method
             )
 
-            # The `file_method` is irrelevant in this situation as the files
-            # are placed directly into the dataset by `record.build()`.
-            dataset.add_file(data_path, labels_path)
+            # MOVE here is used to rename the file
+            dataset.add_file(data_path, labels_path,
+                             os.path.basename(record.new_data_path),
+                             os.path.basename(record.new_labels_path),
```

Comment:
I would recommend passing the final paths directly to `record.build()`. Labels are being loaded, having their `filename` attr changed ,and then written twice in the current form.